src/main/java/de/tum/in/www1/artemis/service/export/ProgrammingExerciseExportService.java

```diff
@@ -141,14 +145,22 @@ private Path exportProgrammingExerciseMaterialWithStudentReposOptional(Programmi
         }
         catch (Exception e) {
             exportErrors.add("Failed to export programming exercise repositories: " + e.getMessage());
-
         }
+
         // Add problem statement as .md file
-        super.exportProblemStatementAndEmbeddedFilesAndExerciseDetails(exercise, exportErrors, exportDir.orElseThrow(), pathsToBeZipped);
+        exportProblemStatementAndEmbeddedFilesAndExerciseDetails(exercise, exportErrors, exportDir.orElseThrow(), pathsToBeZipped);
 
         return exportDir.orElseThrow();
     }
 
+    @Override
+    void exportProblemStatementAndEmbeddedFilesAndExerciseDetails(Exercise exercise, List<String> exportErrors, Path exportDir, List<Path> pathsToBeZipped) throws IOException {
+        if (exercise instanceof ProgrammingExercise programmingExercise) {
```

Comment:
shouldn't this if condition always be fulfilled?
but I think it's fine because otherwise one has to directly cast to a PE.
since you override the method, some javadoc would be nice.

Reply:
Yes, it should always be fulfilled, but I wanted to avoid the unchecked cast.

Reply:
Optional: Add a comment about what you explained here? Otherwise, this info gets lost a bit 🤔